medmodels/medrecord/querying.pyi

```diff
@@ -19,314 +20,377 @@ else:
 NodeQuery: TypeAlias = Callable[[NodeOperand], None]
 EdgeQuery: TypeAlias = Callable[[EdgeOperand], None]
 
-ValueComparisonOperand: TypeAlias = Union[
-    SingleValueOperand, MedRecordValue, MultipleValuesOperand, List[MedRecordValue]
+SingleValueComparisonOperand: TypeAlias = Union[SingleValueOperand, MedRecordValue]
+MultipleValuesComparisonOperand: TypeAlias = Union[
+    MultipleValuesOperand, List[MedRecordValue]
 ]
-ValueArithmeticOperand: TypeAlias = Union[SingleValueOperand, MedRecordValue]
-AttributeComparisonOperand: TypeAlias = Union[
+
+SingleAttributeComparisonOperand: TypeAlias = Union[
     SingleAttributeOperand,
     MedRecordAttribute,
-    MultipleAttributesOperand,
-    List[MedRecordAttribute],
 ]
-AttributeArithmeticOperand: TypeAlias = Union[
-    SingleAttributeOperand, MedRecordAttribute
+MultipleAttributesComparisonOperand: TypeAlias = Union[
+    MultipleAttributesOperand, List[MedRecordAttribute]
 ]
+
+NodeIndexComparisonOperand: TypeAlias = Union[NodeIndexOperand, NodeIndex]
+NodeIndicesComparisonOperand: TypeAlias = Union[NodeIndicesOperand, List[NodeIndex]]
+
 EdgeIndexComparisonOperand: TypeAlias = Union[
     EdgeIndexOperand,
     EdgeIndex,
+]
+EdgeIndicesComparisonOperand: TypeAlias = Union[
     EdgeIndicesOperand,
     List[EdgeIndex],
 ]
-EdgeIndexArithmeticOperand: TypeAlias = Union[EdgeIndexOperand, EdgeIndex]
 
 class EdgeDirection(Enum):
     INCOMING = auto()
     OUTGOING = auto()
     BOTH = auto()
 
 class NodeOperand:
-    def attribute(self, attribute: MedRecordAttribute) -> MultipleValuesOperand: ...
+    def attribute(
+        self, attribute: Union[MedRecordAttribute, SingleAttributeOperand]
+    ) -> MultipleValuesOperand: ...
     def attributes(self) -> MultipleAttributesOperand: ...
     def index(self) -> NodeIndexOperand: ...
     def in_group(self, group: Union[Group, List[Group]]) -> None: ...
     def has_attribute(
-        self, attribute: Union[MedRecordAttribute, List[MedRecordAttribute]]
+        self, attribute: Union[MedRecordAttribute, SingleAttributeOperand]
     ) -> None: ...
     def incoming_edges(self) -> EdgeOperand: ...
     def outgoing_edges(self) -> EdgeOperand: ...
     def neighbors(
         self, edge_direction: EdgeDirection = EdgeDirection.OUTGOING
     ) -> NodeOperand: ...
     def either_or(self, either: NodeQuery, or_: NodeQuery) -> None: ...
+    def clone(self) -> NodeOperand: ...
```

Comment:
What is the purpose of this `clone` operands on Node and Edges? Just curious

Reply:
when you want to save the operand with its current operations and use it later on.

e.g.
```python
def query(node: NodeOperand):
    node.attribute("foo").less_than(5)
    saved_operand = node.clone()
    node.has_attribute("bar")
    ...
    # use `saved_operand` here for other operations where you just want the operand to be filtered with `node.attribute("foo").less_than(5)` and not `has_attribute("bar")`
    ....
```

Reply:
good idea!